### PR TITLE
[MIG][10.0] website_hr_contact

### PR DESCRIPTION
**Pull request by JonathanNEMRY**

### CODE REVIEW

File: setup/website_hr_contact/.eggs/README.txt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+This directory contains eggs that were downloaded by setuptools to build, test, and run plug-ins.
+
+This directory caches those eggs to prevent repeated downloads.
+
+However, it is safe to delete this directory.
+
```
Comment (sbidoul) [inline setup/website_hr_contact/.eggs/README.txt]: remove this file

File: website_hr_contact/README.rst [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+===============
+HR Address Book
+===============
+
+This module add a new page into your website displaying the address book of your employees.
+The informations related to your employees are publicly available if your employee is
+published. The same rule applies to the address informations.
+
+Usage
+=====
```
Comment (yajo) [inline website_hr_contact/README.rst:14]: Please add usage instructions, to know what to test.

**Review by yajo — CHANGES_REQUESTED** [review]

File: website_hr_contact/README.rst (hunk 2 above)
Comment (simahawk) [reply website_hr_contact/README.rst:14]: @JonathanNEMRY ping :)

File: website_hr_contact/README.rst [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/186/8.0
```
Comment (simahawk) [inline website_hr_contact/README.rst:18]: :fire:

File: website_hr_contact/README.rst [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+ * todo: add the department's address book
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/website/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/OCA/website/issues/new?body=module:%20website_hr_employee_addressbook%0Aversion:%208.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (simahawk) [inline website_hr_contact/README.rst:31]: drop this link

File: website_hr_contact/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+# © 2015-2017 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    'name': "Website Hr Contact",
+    'summary': """
+        Display your hr address book in your website
+    """,
+    'author': 'ACSONE SA/NV,'
+              'Odoo Community Association (OCA)',
+    'website': "http://acsone.eu",
```
Comment (simahawk) [inline website_hr_contact/__manifest__.py:11]: https://github.com/OCA/website

File: website_hr_contact/controllers/main.py [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import werkzeug
+
+from odoo import http
+from odoo.addons.website.models.website import slug
+from odoo.osv.orm import browse_record
+
+
+class QueryURL(object):
```
Comment (simahawk) [inline website_hr_contact/controllers/main.py:12]: any reason not to use `from odoo.addons.website.controllers.main import QueryURL`?

File: website_hr_contact/controllers/main.py [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+    def _get_departments_tree(self, department=None, search=None, **post):
+        """ Build a department tree with only published departments
+        The tree only contains published department even if the parent
+        department is not published.
+        """
+        tree = []
+        request = http.request
+        hr_department = request.env['hr.department']
+        sudo_hr_department = hr_department.sudo()
+        visible_departments = hr_department.search([(1, '=', 1)])
```
Comment (simahawk) [inline website_hr_contact/controllers/main.py:70]: why this domain?

File: website_hr_contact/controllers/main.py [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+        departments = sudo_hr_department.search([('parent_id', '=', False)])
+        for department in departments:
+            children = self._get_children_department_recursive(
+                department, visible_departments)
+            if department in visible_departments:
+                tree.append({
+                    'department': visible_departments.filtered(
```
Comment (simahawk) [inline website_hr_contact/controllers/main.py:77]: why do you need this filtering? can't U just use `department`?

File: website_hr_contact/controllers/main.py [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+            department, search, **post)
+
+        domain = self._get_employees_domain(department, search, **post)
+        employees = hr_employee.search(domain)
+
+        url_builder = QueryURL(
+            '/page/hr_contact/employees', ['department', 'page'],
+            department=department)
+
+        count = len(employees)
```
Comment (simahawk) [inline website_hr_contact/controllers/main.py:113]: can't you use search_count and then use limit+offeset to limit the load of the search?

File: website_hr_contact/static/src/css/website_hr_contact.css [hunk 10]
```diff
@@ -0,0 +1,20 @@
+.employees_contact #middle_column>section+section {
+    margin-top: 2em;
+    padding-top: 2em;
+    border-top: 1px solid #ddd;
+}
+.employees_contact dl {
```
Comment (simahawk) [inline website_hr_contact/static/src/css/website_hr_contact.css:6]: useless

File: website_hr_contact/static/src/css/website_hr_contact.css [hunk 11]
```diff
@@ -0,0 +1,20 @@
+.employees_contact #middle_column>section+section {
```
Comment (simahawk) [inline website_hr_contact/static/src/css/website_hr_contact.css:1]: spaces around operators missing

File: website_hr_contact/static/src/css/website_hr_contact.css [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+    clear: left;
+    text-align: right;
+    font-weight: bold;
+}
+.employees_contact dt:after {
+    content: ":\00a0";
+}
+.employees_contact dd {
+    padding: 0 0 0.5em 0;
+}
```
Comment (simahawk) [inline website_hr_contact/static/src/css/website_hr_contact.css:20]: newline missing

File: website_hr_contact/views/website_hr_contact_templates.xml [hunk 13]
```diff
@@ -0,0 +1,191 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+  <data>
```
Comment (simahawk) [inline website_hr_contact/views/website_hr_contact_templates.xml:3]: data not needed

File: website_hr_contact/views/website_hr_contact_templates.xml [hunk 14]
```diff
@@ -0,0 +1,191 @@
...
+      </xpath>
+    </template>
+
+    <template id="employees" name="Employees / Contacts" page="True">
+      <t t-call="website.layout">
+        <div class="container employees_contact">
+          <div class="row mt8 header">
+            <div class="col-sm-7" id="employees_menu">
+                <t t-call="website_hr_contact.search"><t t-set="search_class">pagination form-inline col-md-3</t></t>
+                 <t t-call="website.pager"/>
```
Comment (simahawk) [inline website_hr_contact/views/website_hr_contact_templates.xml:17]: fix indent

File: website_hr_contact/views/website_hr_contact_templates.xml [hunk 15]
```diff
@@ -0,0 +1,191 @@
...
+            </div>
+          </div>
+          <div class="row mt16 mb32">
```
Comment (simahawk) [inline website_hr_contact/views/website_hr_contact_templates.xml:20]: 'employees-listing' class would be nice

File: website_hr_contact/views/website_hr_contact_templates.xml [hunk 16]
```diff
@@ -0,0 +1,191 @@
...
+          <span class="input-group-btn">
+            <a class="btn btn-default a-submit"><i class="fa fa-search"/></a>
+          </span>
+      </div>
+      </form>
+    </template>
+
+    <template id="employee" name="Employee information">
+      <section class="employee_contact">
+      <div class="row">
```
Comment (simahawk) [inline website_hr_contact/views/website_hr_contact_templates.xml:61]: fix indent

File: website_hr_contact/views/website_hr_contact_templates.xml [hunk 17]
```diff
@@ -0,0 +1,191 @@
...
+        <t t-call="website.publish_management">
+          <t t-set="object" t-value="employee"/>
+        </t>
+        <div class="col-md-12" itemscope="itemscope" itemtype="http://schema.org/employee">
+          <section class="row">
+            <div class="col-md-10">
+              <h5 itemprop="name" t-field="employee.name"/>
+              <div class="job_title" itemprop="jobTitle" t-field="employee.job_id"/>
+              <t t-set="department" t-value="employee.department_id.search([('id', '=', employee.department_id.id)])"/>
```
Comment (simahawk) [inline website_hr_contact/views/website_hr_contact_templates.xml:70]: why do you search for something that you have already in the field? :thinking:

File: website_hr_contact/views/website_hr_contact_templates.xml [hunk 18]
```diff
@@ -0,0 +1,191 @@
...
+                    <dt class="email">Email</dt>
+                    <dd class="email"><a itemprop="email" class="link--mail" t-attf-href="mailto:#{employee.work_email}" t-field="employee.work_email"/></dd>
+                </t>
+            </dl>
+        </div>
+    </template>
+
+    <template id="address" name="Address information">
+      <div t-att-class="'address ' + classname" itemprop="address" itemscope="itemscope" itemtype="http://schema.org/PostalAddress">
+        <t t-set="address" t-value="address.search([('id', '=', address.id)])"/>
```
Comment (simahawk) [inline website_hr_contact/views/website_hr_contact_templates.xml:125]: again... you already have "address", why do you search for it?

File: website_hr_contact/views/website_hr_contact_templates.xml [hunk 19]
```diff
@@ -0,0 +1,191 @@
...
+        <t t-if="address.id">
```
Comment (simahawk) [inline website_hr_contact/views/website_hr_contact_templates.xml:126]: if address

**Review by simahawk — CHANGES_REQUESTED** [review]

**Pull request closed without merging**